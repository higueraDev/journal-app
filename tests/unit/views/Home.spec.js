const { shallowMount } = require("@vue/test-utils")
import Home from '@/views/Home'

describe('Pruebas en el Home view', () => { 
    test(('Debe hacer match con el snapshot'),()=>{
        const wrapper = shallowMount(Home);
        expect(wrapper.html()).toMatchSnapshot()
    })
    test('Debe redireccionar a no-entry al hacer click en el boton', () => { 
        const mockRouter = {
            push: jest.fn()
        };

        const wrapper = shallowMount(Home, {
            global:{
                mocks:{
                    $router:mockRouter
                }
            }
        });

        wrapper.find('button').trigger('click')
        expect(mockRouter.push).toHaveBeenCalled()
        expect(mockRouter.push).toHaveBeenCalledWith({name:'no-entry'})
     })
 })