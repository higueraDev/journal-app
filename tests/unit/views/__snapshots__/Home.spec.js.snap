// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Home view Debe hacer match con el snapshot 1`] = `<div class="home"><img alt="Vue logo" src=""><button class="btn btn-primary">Primary</button><button class="btn btn-secondary">Secondary</button><button class="btn btn-success">Success</button></div>`;
